mlflow/projects/__init__.py

```diff
@@ -46,10 +46,20 @@
 _PROJECT_TAR_ARCHIVE_NAME = "mlflow-project-docker-build-context"
 _MLFLOW_DOCKER_TRACKING_DIR_PATH = "/mlflow/tmp/mlruns"
 
-
 _logger = logging.getLogger(__name__)
 
 
+def _resolve_experiment_id(experiment_name=None, experiment_id=None):
+    exp_id = experiment_id
+    if experiment_name:
+        client = tracking.MlflowClient()
+        experiment = client.get_experiment_by_name(experiment_name)
```

Comment:
If experiment_name is set, but the experiment doesn't exist, maybe it would make most sense to throw an error in this case?

Reply:
Similarly, we could disallow setting both experiment_name and experiment_id to non-None values

Reply:
guess this is ```client.get_experiment_by_name``` responsibility to throw exception in case of failure

Reply:
there was a validation logic laying around checking experiment_name and experiment_id both selected. moving it here and adding unit test for clarity

Reply:
Got it, then probably we can avoid having the fallback logic part (currently it looks like it will fallback to use the experiment_id)